lib/src/step/generic_step.dart

```diff
@@ -19,31 +20,22 @@ class GenericStep implements BddStep {
   String get content => '''
 import 'package:flutter_test/flutter_test.dart';
 
-${getStepSignature(rawLine, testerType, customTesterName)} {
+Future<void> $methodName($testerType $customTesterName${_getMethodParameters(rawLine)}) async {
   throw UnimplementedError();
 }
 ''';
 
-  String getStepSignature(
-    String stepLine,
-    String testerType,
-    String testerName,
-  ) {
+  String _getMethodParameters(String stepLine) {
     final params = parseRawStepLine(stepLine).skip(1);
-    if (params.isEmpty) {
-      final examples = examplesRegExp.allMatches(stepLine);
-      if (examples.isEmpty) {
-        return 'Future<void> $methodName($testerType $testerName) async';
-      } else {
-        return _generateSignature(examples.length, testerName);
-      }
+    if (params.isNotEmpty) {
+      return params.mapIndexed((p, index) => ', dynamic param${p + 1}').join();
```

Comment:
It is really a nice improvement for the scenario outline case, and as we are changing the generic step generation, I propose to make life slightly better for regular steps too. For scenario outlines we may detect the name of a parameter, and for other cases we may detect the type like that:
```dart
      return params
          .mapIndexed(
            (index, p) => ', ${_getGenericParameterType(p)} param${index + 1}',
          )
          .join();

...

  String _getGenericParameterType(String parameter) {
    if (parameter == 'true' || parameter == 'false') {
      return 'bool';
    }
    if (num.tryParse(parameter) != null) {
      return 'num';
    }
    if (parameter.startsWith('"') && parameter.endsWith('"') ||
        (parameter.startsWith("'") && parameter.endsWith("'"))) {
      return 'String';
    }
    return 'dynamic';
  }
```

What do you think?